iml-agent/src/crm_reader.rs

```diff
@@ -0,0 +1,158 @@
+// Copyright (c) 2020 DDN. All rights reserved.
+// Use of this source code is governed by a MIT-style
+// license that can be found in the LICENSE file.
+
+use crate::agent_error::ImlAgentError;
+use iml_cmd::{CheckedCommandExt, Command};
+use iml_fs::file_exists;
+use iml_wire_types::high_availability::{Cluster, Node};
+use quick_xml::{
+    events::{attributes::Attributes, Event},
+    Reader,
+};
+use std::{collections::HashMap, convert::TryInto, io};
+
+static CRM_MON_PATH: &'static str = "/usr/sbin/crm_mon";
+
+pub async fn get_crm_mon() -> Result<Option<Cluster>, ImlAgentError> {
+    if !file_exists(CRM_MON_PATH).await {
+        return Ok(None);
+    }
+
+    let x = Command::new(CRM_MON_PATH)
+        .arg("-1")
+        .arg("-r")
+        .arg("-X")
+        .checked_output()
+        .await?;
+
+    let x = read_crm_output(&x.stdout)?;
+
+    Ok(Some(x))
+}
+
+fn required_arg<'a>(arg: &str, x: &'a HashMap<&str, String>) -> Result<&'a str, ImlAgentError> {
+    x.get(arg)
+        .map(|x| x.as_str())
+        .ok_or_else(|| ImlAgentError::MissingArgument(arg.into()))
+}
+
+fn required_bool(arg: &str, x: &HashMap<&str, String>) -> Result<bool, ImlAgentError> {
+    required_arg(arg, x).and_then(str_to_bool)
+}
+
+fn str_to_bool(x: &str) -> Result<bool, ImlAgentError> {
```

Comment:
you could use std::str::FromStr:
```
let b: Result<bool, ParseBoolError> = std::str::FromStr::from_str("true");
```
https://doc.rust-lang.org/std/primitive.bool.html#impl-FromStr

Reply:
Updated to use `.parse`